.github/workflows/main.yml

```
@@ -11,11 +11,16 @@ permissions:
 
 jobs:
   test:
-    name: Test - Python ${{ matrix.python-version }}
-    runs-on: ubuntu-latest
+    name: Test - Python ${{ matrix.python-version }} on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+        python-version: ["3.9", "3.13"]
+        include:
+          - { os: "ubuntu-latest", python-version: "3.10" }
+          - { os: "ubuntu-latest", python-version: "3.11" }
+          - { os: "ubuntu-latest", python-version: "3.12" }
```

I appreciate this change very much, I do the same thing with my packages. I just wonder, why do you limit the python-version to two versions and explicitly add the remaining ones with ubuntu only?

```suggestion
        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
```

---

Premature optimization? My gut feeling tells me that operating specific problems are likely to happen with either all python versions, only late or only new python versions and not in between.
But it is a tradeoff between having Microsoft spend more money on Open Source or a simpler config :-)